lib/mutant/cache.rb

```diff
@@ -21,7 +24,7 @@ def initialize
     # @api private
     def parse(path)
       @cache.fetch(path) do
-        @cache[path] = Parser::CurrentRuby.parse(File.read(path))
+        @cache[path] = Parser::CurrentRuby.parse(file.read(path))
```

Comment:
If you do decide to make `path` a `Pathname` object (as I think you should), this whole `Cache` object becomes unnecessary. You can just use a `ThreadSafe::Cache` instance directly and inject that into the Bootstrap instance.